volume/ceph/ceph.go

```diff
@@ -177,22 +186,56 @@ func (v *Volume) Mount(id string) (mappedDevicePath string, returnedError error)
 	if err != nil {
 		return "", err
 	}
+
+	if fsType == criptoLUKS {
+		cmd = exec.Command("cryptsetup", "luksOpen", v.mappedDevicePath, v.Name())
+		stdin, err := cmd.StdinPipe()
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+		defer stdin.Close()
+		if err := cmd.Start(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		key, err := getLuksKey(v.Name())
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		io.WriteString(stdin, key)
+		io.WriteString(stdin, "\n")
+		if err := cmd.Wait(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+		v.mappedLuksDevicePath = path.Join(LuksDevMapperPath, v.Name())
+	}
+
 	if fsType == "" {
```

Comment:
Before we get here, we need to recheck the fsType, but now checking it from the opened volume. We do want to run fsck etc on the opened cryptovolume.